dali/operators/decoder/audio/audio_decoder_impl.cc

```diff
@@ -17,39 +17,62 @@
 
 namespace dali {
 
-template <typename T>
-span<char> as_raw_span(T *buffer, ptrdiff_t length) {
-  return make_span(reinterpret_cast<char*>(buffer), length*sizeof(T));
+std::pair<int64_t, int64_t> ProcessOffsetAndLength(const AudioMetadata &meta, float offset_sec, float length_sec) {
```

Comment:
```suggestion
std::pair<int64_t, int64_t> ProcessOffsetAndLength(const AudioMetadata &meta, double offset_sec, double length_sec) {
```
With any decent sampling rate, float is not big enough to store the time.

Reply:
Do we expect to have such accuracy?

Reply:
We can also use int64_t (nanoseconds). The capacity is >200 years - much more than enough for audio.